wifiphisher.py

```diff
@@ -881,32 +884,127 @@ def get_hostapd():
     get_hostapd()
 
+    # get all the local templates
+    local_templates = phishingpage.get_local_templates()
+
+    # loop through the local templates
+    for name in local_templates:
+
+        # add the template to the database
+        phishingpage.add_template(name)
+
     # get template_database
     template_database = phishingpage.get_template_database()
 
-    # check to see if the template is local
-    if not template_database[TEMPLATE_NAME] is None:
+    # get all the templtes names for display
+    template_names = list(template_database.keys())
+
+    # variable used to indicate correct template selection
+    template_selected = False
+
+    # loop until all operations for template selection is done
+    while True:
+
+        # check if the template argument is set and is correct
+        if args.template and args.template in template_database:
+
+            # set the template name
+            template_name = args.template
+
+            # set the template selected to avoid interactive selection
+            template_selected = True
+
+        # loop until a valid template is selected
+        while not template_selected:
 
-        # if template is incomplete locally, delete and ask for a download
-        if not phishingpage.check_template(TEMPLATE_NAME):
+            # display start of template names
+            print "\nList Of All Templates: \n"
 
-            # clean up the previous download
-            phishingpage.clean_template(TEMPLATE_NAME)
+            # display the templates
+            for number in range(len(template_names)):
+                print ("[" + G + str(number + 1) + W + "] " +
+                       template_names[number])
+
+            # get user's choice
+            choosen_template = raw_input("Please Select One Of The Above "
+                                         "Templates: ")
+
+            # try to convert the input to integer
+            try:
+
+                template_number = int(choosen_template)
+
+            except ValueError:
+
+                # placed to avoid a program crash in case of non integer input
+                pass
+
+            if (type(template_number) is int and
+                    template_number in range(1, len(template_names) + 1)):
+
+                template_selected = True
+
+            else:
 
-            # get user's response
-            response = raw_input("Template is available online. Do you want"\
-            " to download it now? [y/n] ")
+                print "[" + R + "-" + W + "] Wrong Input! Please Try Again"
+
+            # remove 1 from template number which was added for display reasons
+            template_number -= 1
+
+            try:
+                # set the template name
+                template_name = template_names[template_number]
+            except IndexError:
+
+                # placed to avoid a program crash in case of unexpected
+                # template number
+                pass
```

Comment:
But the script would crash right after on line 1047, right? Instead of `pass` maybe you wanted to put the `continue` statement there.